libbeat/outputs/redis/config.go

```diff
@@ -39,3 +41,17 @@ var (
 		DbTopology:       1,
 	}
 )
+
+func (c *redisConfig) Validate() error {
+	switch c.DataType {
+	case "", "list", "channel":
+	default:
+		return fmt.Errorf("redis data type %v not supported", c.DataType)
+	}
+
+	if c.Index == "" {
+		return errors.New("index required")
```

Comment:
ucfg will wrap all errors inside ucfg.Error adding some more context. ucfg.Error additionall prints the option validation failed for. In this case `output.redis` will be printed in addition.

That's the advantage having errors via Validate, all errors will pass ucfg giving ucfg a chance to add some more context + potentially record all errors (on ucfg roadmap) for printing later.